updater/proxy.go

```diff
@@ -1,41 +1,53 @@
 // SILVER - Service Wrapper
 // Auto Updater
 //
-// Copyright (c) 2014-2021 PaperCut Software http://www.papercut.com/
+// Copyright (c) 2014-2025 PaperCut Software http://www.papercut.com/
 // Use of this source code is governed by an MIT or GPL Version 2 license.
 // See the project's LICENSE file for more information.
 //
 
 package main
 
 import (
-	"io/ioutil"
+	"fmt"
 	"net/http"
 	"net/url"
 	"os"
 	"strings"
 )
 
-func setupHTTPProxy() {
+// setupHTTPProxy attempts to set the HTTP(S)_PROXY vars using
+// the SILVER_HTTP_PROXY or http-proxy.conf file.
+// Return an error if we attempted and failed to do so or the proxy
+// string was not valid.
+func setupHTTPProxy(httpProxyArg string) error {
 	// Force if set via flag
-	if len(*httpProxy) > 0 {
-		_ = os.Setenv("HTTP_PROXY", *httpProxy)
-		return
-	}
-	// Check Silver Environment
-	proxy := os.Getenv("SILVER_HTTP_PROXY")
-	if proxy != "" {
-		_ = os.Setenv("HTTP_PROXY", proxy)
-		return
-	}
-	// Proxy conf file
-	if dat, err := ioutil.ReadFile("http-proxy.conf"); err == nil {
-		proxy = strings.TrimSpace(string(dat))
+	proxy := httpProxyArg
+
+	if proxy == "" {
+		// Else check Silver Environment
+		proxy = os.Getenv("SILVER_HTTP_PROXY")
+		if proxy == "" {
+			// Else check proxy conf file
+			// If conf file is empty of data then there is no proxy set
+			if dat, err := os.ReadFile("http-proxy.conf"); err == nil {
+				proxy = strings.TrimSpace(string(dat))
+			} else {
+				return err
```

Comment:
NIT: It would be better to use early-reaturn.

```go
dat, err := os.ReadFile("http-proxy.conf")
if err != nil {
    return err
}
proxy = strings.TrimSpace(string(dat))
```

Reply:
It would have improved readability if the evaluation had been done from high-priority sources rather than low-priority sources.

1. Use the injected value if not empty and return
2. Use the value in http-prroxy.conf if not empty and return
3. Use SILVER_HTTP_PROXY if not empty and return
4. No proxy